concent_api/core/subtask_helpers.py

```diff
@@ -10,134 +10,173 @@
+            client_public_key=subtask.provider.public_key_bytes,
+            queue=PendingResponse.Queue.Receive,
+            subtask=subtask,
+        )
+        store_pending_message(
+            response_type=PendingResponse.ResponseType.SubtaskResultsSettled,
+            client_public_key=subtask.requestor.public_key_bytes,
+            queue=PendingResponse.Queue.ReceiveOutOfBand,
+            subtask=subtask,
+        )
+    elif subtask.state == Subtask.SubtaskState.VERIFICATION_FILE_TRANSFER.name:  # pylint: disable=no-member
+        update_subtask_state(
+            subtask=subtask,
+            state=Subtask.SubtaskState.FAILED.name,  # pylint: disable=no-member
+        )
+        store_pending_message(
+            response_type=PendingResponse.ResponseType.SubtaskResultsRejected,
+            client_public_key=subtask.provider.public_key_bytes,
+            queue=PendingResponse.Queue.ReceiveOutOfBand,
+            subtask=subtask,
+        )
+        store_pending_message(
+            response_type=PendingResponse.ResponseType.SubtaskResultsRejected,
+            client_public_key=subtask.requestor.public_key_bytes,
+            queue=PendingResponse.Queue.ReceiveOutOfBand,
+            subtask=subtask,
+        )
+    elif subtask.state == Subtask.SubtaskState.ADDITIONAL_VERIFICATION.name:  # pylint: disable=no-member
+        task_to_compute = deserialize_message(subtask.task_to_compute.data.tobytes())
+        # Worker makes a payment from requestor's deposit just like in the forced acceptance use case.
+        payments_service.make_force_payment_to_provider(  # pylint: disable=no-value-for-parameter
+            requestor_eth_address=task_to_compute.requestor_ethereum_address,
+            provider_eth_address=task_to_compute.provider_ethereum_address,
+            value=task_to_compute.price,
+            payment_ts=get_current_utc_timestamp(),
+        )
+
+        update_subtask_state(
+            subtask=subtask,
+            state=Subtask.SubtaskState.ACCEPTED.name,  # pylint: disable=no-member
+        )
+        store_pending_message(
+            response_type=PendingResponse.ResponseType.SubtaskResultsSettled,
+            client_public_key=subtask.provider.public_key_bytes,
+            queue=PendingResponse.Queue.ReceiveOutOfBand,
+            subtask=subtask,
+        )
+        store_pending_message(
+            response_type=PendingResponse.ResponseType.SubtaskResultsSettled,
+            client_public_key=subtask.requestor.public_key_bytes,
+            queue=PendingResponse.Queue.ReceiveOutOfBand,
+            subtask=subtask,
+        )
+
+    log_string_message(
+        logger,
+        f"Subtask with id: {subtask.subtask_id} changed it's state from: {subtasks_initial_state} to: {subtask.state}. "
+        f"Provider id: {subtask.provider_id}. Requestor id: {subtask.requestor_id}."
+    )
+
+
+def update_timed_out_subtasks_in_message(client_message: Message, client_public_key: bytes) -> None:
```

Comment:
This function really should have a better name or at least a docstring that says what it's supposed to do.

Reply:
sorry for this, didn't saw your comment before merge. I will fix it today in other PR.

Reply:
I added it after merge :)

I wanted to review this pull request but it was merged before I managed to so - so I just took a brief look now. I did not find any problems and this is the only thing that stood out to me. There are two very similarly named functions and one checks upload status and the other does not but the name does not suggest such a difference at all.

@dybi @rwrzesien As a general observation - when a function has a good, descriptive name, docstring may be superfluous. But in situations like this where you have trouble finding a good name I think it's a good idea to have a docstring that explains it more verbosely.

Reply:
Good naming is hard and comes with experience :) I think in this case docstrings are not necessary, I tried to make descriptive names in views:
`update_subtasks_from_incomming_message_if_timed_out` - send
`update_all_timed_out_subtasks_of_client` - receive
And both this functions select subtasks to update and call `_update_timed_out_subtask`  
Is it OK?

Reply:
It's still confusing because by looking at the names of `update_subtasks_from_incomming_message_if_timed_out()` and `_update_timed_out_subtask()` I would never suspect that the first one checks if files have been uploaded and the other does not. It also does not indicate that one locks the subtask and the other does not.

If you can't or don't want include this information in the name, you should put it in a docstring.

BTW,
- `incomming` -> `incoming`
- `of_client` -> `of_a_client`

Reply:
Let's move this discussion to https://github.com/golemfactory/concent/pull/841. I will fix it here. I tried to make good names, but if it is still confusing I will rename it again and add docstring. Everything should be obvious :)